src/transformers/models/sam/processing_sam.py

```diff
@@ -33,6 +34,23 @@
     import tensorflow as tf
 
 
+class SamImagesKwargs(ImagesKwargs):
+    segmentation_maps: Optional[ImageInput]
```

Comment:
I think it this `segmentation_maps` is not required for the inference of SAM.

Reply:
This is needed for backwards compatibility - even if it is not used. E.g. take the following call: `processor(images, None, input_points)`. If I would remove it from `_add_args_for_backward_compatibility`, such calls would break.
If I understood correctly, #31911 is about uniform kwargs, and then at a later step the API would be cleaned up further. In SAM's case, `_add_args_for_backward_compatibility` will be removed, and then `segmentation_maps` can be removed as well.

Reply:
We actually have an existing way to handle such positional args, you can take a look at [udop processor](https://github.com/huggingface/transformers/blob/main/src/transformers/models/udop/processing_udop.py) for example :)